code/game/machinery/autolathe.dm

```diff
@@ -228,14 +228,21 @@
 	else
 		for(var/i=1, i<=multiplier, i++)
 			var/obj/item/new_item = new being_built.build_path(A)
-			new_item.autolathe_crafted(src)
 
 			if(length(picked_materials))
 				new_item.set_custom_materials(picked_materials, 1 / multiplier) //Ensure we get the non multiplied amount
 				for(var/x in picked_materials)
 					var/datum/material/M = x
 					if(!istype(M, /datum/material/glass) && !istype(M, /datum/material/iron))
 						user.client.give_award(/datum/award/achievement/misc/getting_an_upgrade, user)
+			else
+				// we STILL need to set custom materials or we get unlimited autolathe works happening
+				var/old_material_flags = new_item.material_flags
```

Comment:
this entire "set the flag then set the custom materials then unset the flags" makes no sense. Why would you temporary make materials not have effects then unset it? that's just going to make it confusing to work with.

Reply:
I know, but...humm

I was going to argue the point that some items change their specs depending on the materials being used, but now that I think about it, it doesn't matter does it.  In fact, we don't even need to have two separate "material" lists go to this function.  The flags are just there so the user can choose what materials are used.  Let me take a look at it again.